x-pack/plugins/fleet/server/constants/mappings.ts

```diff
@@ -295,6 +295,47 @@ export const AGENT_MAPPINGS = {
     upgrade_status: {
       type: 'keyword',
     },
+    upgrade_details: {
```

Comment:
Adding this comment based on our offline discussion.
This mappings file is there for the only purpose of helping in the KQL validation. It's not adding any new field to the agent index mappings as those live in [Elasticsearch repo](x-pack/plugin/core/template-resources/src/main/resources/fleet-agents.json). I think you'll need to open a separate PR there.

Also, this makes me think that the comment I had on this file is not super clear, so I'll open a small PR to better clarify what's the purpose of this file.

However this is a nice addition because it will enable the KQL searchbox to search on the newly defined fields.

Reply:
I just checked the existing mappings and noticed that they were added already in this PR https://github.com/elastic/elasticsearch/pull/98680. So you should be good there :)

Reply:
Thank you for checking this @criamico! 👍